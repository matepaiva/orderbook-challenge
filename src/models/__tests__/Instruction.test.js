const { Document } = require('camo');
const { Instruction } = require('../index');

jest.mock('../modifiers', () => ({
    withTimestamps: (val) => val,
}));

jest.mock('camo', () => ({
    Document: class MockDocument { },
}));


describe('Instruction', () => {
    describe('.constructor()', () => {
       it('should define fields', () => {
           const instruction = new Instruction();

           expect(instruction).toMatchSnapshot();
           expect(instruction).toBeInstanceOf(Document);
       });
    });

    describe('.collectionName()', () => {
        it('should return the collection name', () => {
            expect(Instruction.collectionName()).toBe('instructions');
        });
    });
});
