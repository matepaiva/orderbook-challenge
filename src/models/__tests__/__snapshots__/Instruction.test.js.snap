// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instruction .constructor() should define fields 1`] = `
Instruction {
  "amount": Object {
    "required": true,
    "type": [Function],
  },
  "orderbookId": Object {
    "required": true,
    "type": [Function],
  },
  "price": Object {
    "min": 0,
    "required": true,
    "type": [Function],
  },
  "quantity": Object {
    "required": true,
    "type": [Function],
  },
  "sourceCurrency": Object {
    "required": true,
    "type": [Function],
  },
  "targetCurrency": Object {
    "required": true,
    "type": [Function],
  },
  "weight": Object {
    "max": 1,
    "min": 0,
    "required": true,
    "type": [Function],
  },
}
`;
